### ISSUE TITLE
Short lesson with repo2docker only for CodeRefinery workshop

### ISSUE THREAD

**Issue by annefou** [coderefinery/jupyter#39]
It is a nice lesson but cannot be taught in 1 or 2 hours. 
Second time I see this lesson (last time was in Goteborg) and I have the same feeling. We have it at the workshop (everybody is tired and we have no more than 1 hour for this lesson, including exercises) but do not have a clear idea on how this tool fits into the Software development framework.

I suggest we make a 1/1.5 hour lesson:

- Teach this lesson after Reproducible Research or documentation (on Wednesday morning) and show how to use jupyter notebooks to document/publish worflows/examples using repo2docker (binder)
- Have one single exercise where using binder is the main objective.

We could also add links to examples such as (these are from Geosciences but you probably have good examples too!):
- https://github.com/binder-examples
- https://github.com/OpenGeoVis/omfvista
- https://github.com/K3D-tools/K3D-jupyter
- https://github.com/jupyter-widgets/ipyleaflet
- https://github.com/pangeo-data/pangeo-example-notebooks

Keep a separate 1/2 day lesson on Jupyter ecosystem we can use as standalone lesson (we would use it in Oslo for 1 day workshop).

What do you think?

**Comment by bast** [comment]
I have the same feeling watching this lesson a couple of times and I like the suggestion..

**Comment by bast** [comment]
Same feeling still. I think that the binder part may be for me the most important outcome and I will focus on that. I will teach this lesson coming Tuesday and maybe manage to readjust the lesson until then.

**Comment by bast** [comment]
I think this part could be removed: https://coderefinery.github.io/jupyter/05-exercises/
It's interesting, but it's more about Python than it is about Jupyter. All in there is nice to have but IMO not a must have. If I was a beginner and to be convinced about Jupyter and perhaps a beginner in Python, this part would be overwhelming for me. We often do not have really the time for this part. If I would remove one part, I would remove this part.

Maybe the one part I would keep is the widget example.

**Comment by bast** [comment]
Instead of showing all the cool things that can be done in Python and Jupyter, for me the goal is more to convince whether and how to move own output->data->plot scripts from the harddrive and ad-hoc scripts to Jupyter+Binder. That's the learning outcome for me. I might spin-off a side-repo where I condense the lesson.

**Comment by wikfeldt** [comment]
what about moving selected parts of this lesson into the reproducibility lesson? we can then retire the jupyter lesson from regular workshops but keep it for special occasion

**Comment by bast** [comment]
I don't think we should retire the Jupyter lesson. I think it is still relevant. The Binder part is new to most. But it could become part of the reproducibility lesson. We can also keep the exercises as optional for special occasions.

**Comment by annefou** [comment]
binder and repo2docker are more general than jupyter. I think it would be better to discuss it during the reproducibilty lesson. That's something we can discuss during our first Open House!

**Comment by bast** [comment]
Moving this good suggestion to reproducibility lesson. Which doesn't mean it needs to be implemented immediately. It has been dormant here for 1.5 years.
